builtin_set.go

```diff
@@ -344,3 +344,90 @@ func (r *Runtime) getSet() *Object {
 	}
 	return ret
 }
+
+// estimateMemUsage helps calculating mem usage for large objects.
+// It will sample the object and use those samples to estimate the
+// mem usage.
+func (so *setObject) estimateMemUsage(ctx *MemUsageContext) (estimate uint64, err error) {
+	var samplesVisited, memUsage uint64
+	totalItems := so.m.size
+	if totalItems == 0 {
+		return memUsage, nil
+	}
+	sampleSize := ctx.ComputeSampleStep(totalItems)
+
+	// We can use samplesVisited instead of an index since we iterate using
+	// iterNext
+	for item := so.m.iterFirst; item != nil && samplesVisited < uint64(totalItems); item = item.iterNext {
```

Comment:
[q] looks like totalItems is the size of the map but we only increment `sampleVisited` when its divisible by the sample size which means this case should never be true `samplesVisited < uint64(totalItems)` (unless you are accounting for size of 1?). Can we remove that check `samplesVisited < uint64(totalItems)`:

```
	for item := so.m.iterFirst; item != nil; item = item.iterNext {
```

Reply:
you're right, even with a step of 1 we would have a nil item and get out I believe so this is not needed.